.travis.yml

```diff
@@ -1,15 +1,53 @@
-branches:
-  only:
-    - master
-    - develop
+sudo: false
 language: node_js
 node_js:
   - "8"
-sudo: required
-install:
-  - yarn
-  - sudo pip install proselint
-before_script:
-  - source ./src/scripts/env.sh
-script:
-  - bash ./src/scripts/deploy.sh
+
+cache:
+  yarn: true
+  pip: true
+  directories:
+    - ./node_modules
+    - ./dist
+    - $HOME/.cache
+
+stages:
+  - lint
+  - build
+  - post-build
+
+jobs:
+  fast_finish: true
+  allow_failures:
+    - name: Link check
+
+  include:
+    - stage: lint
+      name: Linting
+      before_install: npm install --global yarn
+      install: yarn
+      script:
+        - yarn lint:js
+        - yarn lint:markdown
+        - yarn lint:social
+
+    - stage: build
+      name: Build
+      before_script: source ./src/scripts/env.sh
+      script: yarn build
+
+    - stage: post-build
+      name: Deploy
+      if: branch = master
+      script: bash ./src/scripts/deploy.sh
+
+    - stage: post-build
+      name: Link check
+      script: yarn linkcheck
+
+    - stage: post-build
```

Comment:
Why it this one post build? We shouldn't build if this command fails.

Reply:
We can re-order as we see fit later. Right now I just need to get the pieces working, then we can discuss what the pipeline should do.